subsys/shell/shell.c

```diff
@@ -762,6 +793,10 @@ static void tab_handle(const struct shell *shell)
 	size_t argc;
 	size_t cnt;
 
+	if (!IS_ENABLED(CONFIG_SHELL_TAB)) {
```

Comment:
I guess this place is not needed, because `tab_handle` is called only in:
```
				if (flag_echo_get(shell) &&
				    IS_ENABLED(CONFIG_SHELL_TAB)) {
					/* If the Tab key is pressed, "history
					 * mode" must be terminated because
					 * tab and history handlers are sharing
					 * the same array: temp_buff.
					 */
					flag_history_exit_set(shell, true);
					tab_handle(shell);
				}
```

Reply:
I've just tested it. Keeping it saves 16Bytes. The compiler is configured for size optimization.

Reply:
This is weird. I've checked that quickly now and I see no code size difference when removed `if (!IS_ENABLED(CONFIG_SHELL_TAB)) {` in `tab_handle`. Also compiled with size optimizations.

Reply:
There shall be no `!` in the if

Reply:
What do you mean? There is `!` in the source code line we are commenting here.

Reply:
I am referring to your comment:

` if (!IS_ENABLED(CONFIG_SHELL_TAB)) {`

It has `!` while it shall not.

Reply:
Just to be clear, I was referring to if-statement in body of `tab_handle()`. It has `!` in your proposed changes and it should.

And I was proposing to remove
```
	if (!IS_ENABLED(CONFIG_SHELL_TAB)) {
		return;
	}
```
from `tab_handle()`.

If you were talking about
```
if (flag_echo_get(shell) &&
    IS_ENABLED(CONFIG_SHELL_TAB)) {
```
in `state_collect()`, then I have no objections there. It should be like it is.